app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingRunnable.java

```diff
@@ -208,6 +212,49 @@ private boolean useWakeLock() {
         return mPreferences.getBoolean(Constants.PREF_USE_WAKE_LOCK, false);
     }
 
+    /**
+     * Look for running libsyncthing.so processes and store the PID of each instance.
+     */
+    private void getSyncthingPIDs() {
+        Process ps = null;
+        DataOutputStream psOut = null;
+        BufferedReader br = null;
+        mLibSyncthingPID.clear();
+        try {
+            ps = Runtime.getRuntime().exec((mUseRoot) ? "su" : "sh");
+            psOut = new DataOutputStream(ps.getOutputStream());
+            psOut.writeBytes("ps\n");
+            psOut.writeBytes("exit\n");
+            psOut.flush();
+            ps.waitFor();
+            br = new BufferedReader(new InputStreamReader(ps.getInputStream(), "UTF-8"));
+            String line;
+            while ((line = br.readLine()) != null) {
+                if (line.contains(Constants.FILENAME_SYNCTHING_BINARY)) {
+                    String syncthingPID = line.trim().split("\\s+")[1];
+                    Log.v(TAG, "getSyncthingPIDs: Found process PID [" + syncthingPID + "]");
+                    mLibSyncthingPID.add(syncthingPID);
+                }
+            }
+        } catch (IOException | InterruptedException e) {
+            Log.w(TAG_KILL, "Failed list Syncthing processes", e);
+        } finally {
+            try {
+                if (br != null) {
+                    br.close();
+                }
+                if (psOut != null) {
+                    psOut.close();
+                }
+            } catch (IOException e) {
+                Log.w(TAG_KILL, "Failed close the psOut stream", e);
+            }
+            if (ps != null) {
+                ps.destroy();
+            }
+        }
+    }
+
```

Comment:
Why can't this just return the pid list? Whats the purpose of storing it on the class?

Reply:
I first thought of we could store our running syncthing instances on the class and later provide an interface "isBinaryRunning". This was when I discovered another issue (see other PR not relevant here)... but as this is unrelated to the duplicate starts of the binary, you're right. I'll improve that for simplicity :).